### DISCUSSION TITLE
Are metrics timestamped at the start of the scrape, or at the end?

### DISCUSSION THREAD

**Discussion by ruuda** [prometheus/prometheus#11261] [Q&A] [answered]
Suppose my `/metrics` endpoint is kind of slow, and it takes 15 seconds after sending the request, before the server begins sending the response. If Prometheus sends the request at time _t_, and finishes reading the response at time _t_ + 15, what timestamp will Prometheus assign to the collected samples? _t_ or _t_ + 15? Or something in between?

If the answer is _t_ + 15, how does that change when the server is quick to respond with http headers, but slow to finish writing the body? E.g. maybe Prometheus reads the `200 OK` at time _t_ + 1, but it only reads the final byte of the body at time _t_ + 15?

**Comment by roidelapluie** [comment, marked as answer]
The timestamp is the time Prometheus makes the request, so the start of
the scrape.


-- 
Julien Pivotto
O11y - https://o11y.eu/
